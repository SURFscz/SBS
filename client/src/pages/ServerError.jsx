import React from "react";
import I18n from "i18n-js";
import "./ServerError.scss";
import {Ghost} from 'react-kawaii'

export default function ServerError() {
    return (
        <div className="mod-server-error">
            <div className="title">
                <p>{I18n.t("error_dialog.body")}</p>
            </div>
            <div className="content">
                <Ghost size={160} mood="ko" color="#fbfbfb" />
            </div>
        </div>
    );
}