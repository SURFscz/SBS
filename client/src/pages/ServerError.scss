@import "../stylesheets/vars.scss";

.mod-server-error {

    background-color: $background;
    margin-bottom: $medium-padding;

    div.title {
        font-size: 20px;
        color: $dark-grey;
        max-width: $medium;
        margin: 20px auto;
        padding: 10px 0 5px 0;

        p {
            padding: 10px 0 10px 0;
        }

    }

    .content {
        max-width: $medium;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
        margin: 0 auto 25px auto;
        background-color: white;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        padding: 80px;

    }
}
