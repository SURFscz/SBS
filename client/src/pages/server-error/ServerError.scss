@import "../../stylesheets/vars";

.mod-server-error {
    max-width: $medium;
    margin: 0 auto;
    width: 100%;

    .content {
        margin: 0 auto 25px auto;
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        padding: 15px;
        p {
            &.status {
                font-family: "Courier New", serif;
                margin: 15px 0;
                font-weight: bold;
            }
        }
        svg {
            margin-top: 25px;
            width: 480px;
            height: auto;

        }
    }
}
