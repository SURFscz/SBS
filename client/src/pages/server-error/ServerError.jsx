import React, {useEffect, useState} from "react";
import "./ServerError.scss";
import {ReactComponent as Warning} from "../../lotties/undraw_warning.svg";
import I18n from "../../locale/I18n";
import DOMPurify from "dompurify";
import {isEmpty} from "../../utils/Utils";

export default function ServerError() {

    const [error, setError] = useState(I18n.t("error.message"));
    const [code, setCode] = useState("");
    const [message, setMessage] = useState("");

    useEffect(() => {
        const urlSearchParams = new URLSearchParams(window.location.search);
        const reason = urlSearchParams.get("reason");
        const errorTranslation = I18n.translations[I18n.locale].error[reason];
        if (reason && errorTranslation) {
            setError(errorTranslation);
            const samlCode = urlSearchParams.get("code");
            if (samlCode) {
                setCode(samlCode);
            } else {
                setCode(I18n.t("error.defaultCode"));
            }
            const samlMsg = urlSearchParams.get("msg");
            if (samlMsg) {
                setMessage(samlMsg);
            } else {
                setMessage(I18n.t("error.defaultMessage"));
            }
        }
    }, [])

    return (
        <div className="mod-server-error">
            <div className="content">
                {<p dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(error)}}/>}
                {!isEmpty(code) &&
                    <p className={"status"} dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(code)}}/>}
                {!isEmpty(message) &&
                    <p className={"status"} dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(message)}}/>}
                <Warning/>
            </div>
        </div>
    );
}
