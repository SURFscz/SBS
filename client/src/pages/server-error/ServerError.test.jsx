import React from "react";
import { render } from "@testing-library/react";
import ServerError from "./ServerError";

describe("ServerError", () => {
    it("matches snapshot with default props", () => {
        const { asFragment } = render(
            <ServerError />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
