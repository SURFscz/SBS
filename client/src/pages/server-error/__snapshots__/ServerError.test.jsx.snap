// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerError matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="mod-server-error"
  >
    <div
      class="content"
    >
      <p>
        We seem to have run into an unknown error. Sorry for that.
        <br />
        Start over on the 
        <a
          href="/"
        >
          homepage
        </a>
         or contact us at 
        <a
          href="mailto:sram-support@surf.nl"
        >
          sram-support@surf.nl
        </a>
        .
      </p>
      <svg>
        undraw_warning.svg
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
